### PR TITLE
Aurora support

### PR DESCRIPTION
**Pull request by MattiasGees**
This will add aurora support as database part.

### CODE REVIEW

File: aurora/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output "aurora_port" {
+  value = "3306"
+}
+
```
Comment (simonrondelez) [inline aurora/outputs.tf:4]: maybe output the dns name aswell?

File: README.md [hunk 2]
```diff
@@ -46,3 +46,39 @@ module "rds" {
...
+* [`password`]: String(optional) RDS root password
+* [`backup_retention_period`]: int(optional) How long do you want to keep RDS backups (default: 14)
+* [`apply_immediately`]: bool(optional) whether you want to Apply changes immediately (default: true)
+* [`storage_encrypted`]: bool(optional) whether you want to Encrypt RDS storage (default: true)
+* [`tag`]: String(optional) tag
+* [`project`]: String(required) the name of the project this RDS belongs to
+* [`environment`]: String(required) the name of the environment these subnets belong to (prod,stag,dev)
+* [`skip_final_snapshot`]: bool(optional) Whether to skip creating a final snapshot when destroying the resource (default: false)
+* [`rds_parameter_group_name`]: String(required) the parameter group that is used for the db (supported: `mysql-rds-${var.project}-${var.environment}${var.tag}`, `oracle-rds-${var.project}-${var.environment}${var.tag}`,`postgres-rds-${var.project}-${var.environment}${var.tag}`)
+* [`amount_of_instances`]: Integer(optional) How many aurora instances do you need, minumum 2 are needed for HA (default: 1)
```
Comment (simonrondelez) [inline README.md]: minimum 2 default 1?

File: aurora/main.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  parameter = {
+    name = "log_output"
+
+    value = "FILE"
+  }
+}
+
+resource "aws_rds_cluster" "aurora" {
+  cluster_identifier              = "${var.project}-${var.environment}${var.tag}-aurora"
+  master_username                 = "root"
```
Comment (simonrondelez) [inline aurora/main.tf]: why not a var for this?

File: README.md [hunk 4]
```diff
@@ -46,3 +46,39 @@ module "rds" {
...
 }
 ```
+## Aurora
+Creates a Aurora cluster + instances, security_group, subnet_group and parameter_group
+
+### Available variables:
+* [`security_groups`]: List(required) Security groups that are allowed to access the RDS
+* [`subnets`]: List(required) Subnets to deploy the RDS in
+* [`size`]: String(optional) RDS instance size
+* [`password`]: String(optional) RDS root password
```
Comment (simonrondelez) [inline README.md]: required as i can see in variables.tf

File: aurora/main.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  parameter = {
+    name = "log_output"
+
+    value = "FILE"
+  }
+}
+
+resource "aws_rds_cluster" "aurora" {
```
Comment (ringods) [inline aurora/main.tf:41]: With the upcoming PostgreSQL support for Aurora, can you allow the `engine` and `engine_version` to be specified too?

https://aws.amazon.com/blogs/aws/amazon-aurora-update-postgresql-compatibility/

**Review by ringods — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (MattiasGees) [inline README.md]: if you want HA you need 2. None HA is 1.

File: aurora/main.tf (hunk 3 above)
Comment (MattiasGees) [inline aurora/main.tf]: Took that over from the rds module. root seems like a good fit to me for it.

File: aurora/main.tf (hunk 5 above)
Comment (MattiasGees) [reply aurora/main.tf:41]: This is not yet in the terraform resource.

Comment (MattiasGees) [reply aurora/main.tf:41]: You can check the Terraform docs here:
https://www.terraform.io/docs/providers/aws/r/rds_cluster.html and https://www.terraform.io/docs/providers/aws/r/rds_cluster_instance.html

Comment (ringods) [reply aurora/main.tf:41]: What exactly is not in the terraform resource? The parameters I refer to are available according the docs:
https://www.terraform.io/docs/providers/aws/r/rds_cluster.html#engine

File: aurora/variables.tf [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  default     = false
+}
+
+variable "amount_of_instances" {
+  description = "The amount of Aurora instances you need, for HA you need minumum 2"
+  default     = 1
+}
+
+variable "rds_parameter_group_name" {
+  description = "Optional paramter group you can set for the RDS cluster "
```
Comment (ringods) [inline aurora/variables.tf]: Typo: "paramter"

File: aurora/main.tf (hunk 5 above)
Comment (MattiasGees) [reply aurora/main.tf:41]: @ringods that is an output not an input.

File: aurora/variables.tf [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  default     = ""
+}
+
+variable "tag" {
+  description = "A tag used to identify an RDS in a project that has more than one RDS"
+  default     = ""
+}
+
+variable "skip_final_snapshot" {
+  description = "Skip final snapshot when destorying RDS"
```
Comment (ringods) [inline aurora/variables.tf]: Typo: destorying

File: aurora/main.tf (hunk 5 above)
Comment (ringods) [reply aurora/main.tf:41]: Indeed. I should pay more attention. :-)

Comment (MattiasGees) [reply aurora/main.tf:41]: I looked for it trust me to add multiple engine support :)

File: README.md [hunk 8]
```diff
@@ -46,3 +46,39 @@ module "rds" {
...
+* [`password`]: String(required) RDS root password
+* [`backup_retention_period`]: int(optional) How long do you want to keep RDS backups (default: 14)
+* [`apply_immediately`]: bool(optional) whether you want to Apply changes immediately (default: true)
+* [`storage_encrypted`]: bool(optional) whether you want to Encrypt RDS storage (default: true)
+* [`tag`]: String(optional) tag
+* [`project`]: String(required) the name of the project this RDS belongs to
+* [`environment`]: String(required) the name of the environment these subnets belong to (prod,stag,dev)
+* [`skip_final_snapshot`]: bool(optional) Whether to skip creating a final snapshot when destroying the resource (default: false)
+* [`rds_parameter_group_name`]: String(required) the parameter group that is used for the db (supported: `mysql-rds-${var.project}-${var.environment}${var.tag}`, `oracle-rds-${var.project}-${var.environment}${var.tag}`,`postgres-rds-${var.project}-${var.environment}${var.tag}`)
+* [`amount_of_instances`]: Integer(optional) How many aurora instances do you need, minumum 2 are needed for HA (default: 1)
```
Comment (duboisph) [inline README.md]: *minimum

File: aurora/outputs.tf (hunk 1 above)
Comment (MattiasGees) [reply aurora/outputs.tf:4]: done

**Review by ringods — APPROVED** [review]

**Review by duboisph — APPROVED** [review]

**Pull request merged**